src/main/java/sparksoniq/jsoniq/compiler/translator/expr/primary/VariableReference.java

```diff
@@ -38,6 +38,12 @@
     public VariableReference(String _name, ExpressionMetadata metadata) {
         super(metadata);
         this.name = _name;
+        // TODO: can we eliminate this assignment by properly visiting VariableReferences in staticContextVisitor
+        _highestExecutionMode = ExecutionMode.LOCAL;
```

Comment:
Removing this line causes a multitude of runtime exceptions as VariableReferenceIterators(created from variableReferences that inherit from Expression) have their execution mode unset. This is caused by them not being properly visited by the staticContextVisitor which sets the execution mode by looking up the variable in the context. I think this will require a bit of work on the staticContextVisitor if we wish to address this.

Reply:
This is addressed